masonry/src/widget/widget_pod.rs

```diff
@@ -41,17 +46,29 @@ impl<W: Widget> WidgetPod<W> {
 
     /// Create a new widget pod with fixed id.
     pub fn new_with_id(inner: W, id: WidgetId) -> WidgetPod<W> {
+        Self::new_with_id_and_transform(inner, id, Affine::IDENTITY)
+    }
+
+    /// Create a new widget pod with a custom transform.
+    pub fn new_with_transform(inner: W, transform: Affine) -> WidgetPod<W> {
+        Self::new_with_id_and_transform(inner, WidgetId::next(), transform)
+    }
```

Comment:
Okay, so I get why the PR does it this way, but I'm somewhat uncomfortable with adding a new constructor and all the special-casing that comes with it for a single attribute.

I don't know how else to do it, ideally it should be part of the styling refactor, but I don't want to block on that.

Reply:
Yes I don't think this should be a long term solution, what we could also do instead is something like this:

```rust
    /// Create a new widget pod with fixed id.
    pub fn new_with_id(inner: impl Into<CreatedWidget<W>>, id: WidgetId) -> WidgetPod<W> {
        Self::new_with_id_and_transform(inner, id, Affine::IDENTITY)
    }
```

and add the relevant `From` impls, like e.g. this:

```rust
impl<W: Widget> From<W> for CreatedWidget<W>{..}
impl<W: Widget> From<(W, Affine)> for CreatedWidget<W>{..}
```

for general attributes.
As mentioned somewhere else we could also introduce a builder type for creating widgets, maybe by just using `CreatedWidget` for this?